library/ssl_tls13_generic.c

```diff
@@ -5011,48 +5019,48 @@ static int ssl_new_session_ticket_parse( mbedtls_ssl_context* ssl,
                      ( (unsigned) buf[6] << 8  ) |
                      ( (unsigned) buf[7] << 0  );
 
-    MBEDTLS_SSL_DEBUG_MSG( 3, ( "ticket->age_add: %u", ssl->session->ticket_age_add ) );
+    MBEDTLS_SSL_DEBUG_MSG( 3, ( "ticket->ticket_age_add: %u", ssl->session->ticket_age_add ) );
 
     /* Ticket Nonce */
-    ticket_nonce_len = buf[8];
-
-    MBEDTLS_SSL_DEBUG_MSG( 3, ( "ticket->nonce_length: %d", ticket_nonce_len ) );
+    ssl->session->ticket_nonce_len = buf[8];
 
-    used += ticket_nonce_len;
+    used += ssl->session->ticket_nonce_len;
 
     if( used > buflen )
     {
          MBEDTLS_SSL_DEBUG_MSG( 1, ( "bad new session ticket message" ) );
          return( MBEDTLS_ERR_SSL_BAD_HS_NEW_SESSION_TICKET );
     }
 
-    MBEDTLS_SSL_DEBUG_BUF( 3, "ticket->nonce:", (unsigned char*)&buf[9], ticket_nonce_len );
-
     /* Check if we previously received a ticket already. If we did, then we should
      * re-use already allocated nonce-space.
      */
-    if( ssl->session->ticket_nonce != NULL || ssl->session->ticket_nonce_len > 0 )
+    if( ssl->session->ticket_nonce != NULL && ssl->session->ticket_nonce_len > 0 )
     {
         mbedtls_free( ssl->session->ticket_nonce );
         ssl->session->ticket_nonce = NULL;
         ssl->session->ticket_nonce_len = 0;
     }
 
-    if( ticket_nonce_len > 0 )
+    if( ssl->session->ticket_nonce_len > 0 )
     {
-        if( ( ssl->session->ticket_nonce = mbedtls_calloc( 1, ticket_nonce_len ) ) == NULL )
+        if( ( ssl->session->ticket_nonce = mbedtls_calloc( 1,
+            ssl->session->ticket_nonce_len ) ) == NULL )
         {
             MBEDTLS_SSL_DEBUG_MSG( 1, ( "ticket_nonce alloc failed" ) );
             return( MBEDTLS_ERR_SSL_ALLOC_FAILED );
         }
 
-        memcpy( ssl->session->ticket_nonce, &buf[9], ticket_nonce_len );
-    }
+        memcpy( ssl->session->ticket_nonce, &buf[9], ssl->session->ticket_nonce_len );
 
-    ssl->session->ticket_nonce_len = ticket_nonce_len;
+        MBEDTLS_SSL_DEBUG_BUF( 3, "ticket->nonce:", (unsigned char*)&buf[9], 
+        ssl->session->ticket_nonce_len );
+
+    }
 
     /* Ticket */
-    ticket_len = ( buf[9+ ticket_nonce_len] << 8 ) | ( buf[10+ ticket_nonce_len] );
+    ticket_len = ( buf[9 + ssl->session->ticket_nonce_len] << 8 ) | 
```

Comment:
`ssl->session->ticket_nonce_len` may be set to 0 in L5042. `ssl->session->ticket_nonce_len` will be 0 here in new logic, it will be different to old logic of using a local variable,`ticket_nonce_len`.